File: BAEKJOON/Dec/1212-2775.py
# https://www.acmicpc.net/problem/2775

'''
a층의 b호에 살려면 자신의 아래(a-1)층의 1호부터 b호까지 사람들의 수의 합만큼 사람들을 데려와 살아야 한다
2층, 3호
1 | 2 | 3
1 | 1+2 | 1 + 2+3
1 | 1 + 1+2 | 1 + 1+2 + 1+2+3 
'''

T = int(input())

for _ in range(T):
    k = int(input())
    n = int(input())
    
    # 호수를 키, 사람 수를 값으로 하는 딕셔너리 생성
    floor={}
    
    # 0층 업데이트
    for i in range(1,n+1):
        floor[i] = i
     
    # ex) {1 : 1, 2 : 2, 3 : 3}   
        
    # 그 위의 층
    for i in range(k):    
        # 1호는 어차피 1명이므로 2호부터 시작, 1호부터 하면 j-1에서 에러 발생
        for j in range(2,n+1):
            floor[j] = floor[j-1] + floor[j]
            
    # ex) {1 : 1, 2 : 4, 3 : 10}

    print(floor[n])
        

File: BAEKJOON/Dec/1212-2775.java
/* https://www.acmicpc.net/problem/2775 */

import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        int T = scanner.nextInt();

        for (int t = 0; t<T; t++){
            int k = scanner.nextInt();
            int n = scanner.nextInt();

            // 1행, n+1열(0층부터 시작하므로) 크기의 이차원 배열 생성 
            int[][] floor = new int[1][n + 1];

            // 0층 업데이트
            for (int i = 1; i <= n; i++) {
                floor[0][i] = i;

            }

            // 그 위의 층
            for (int i = 1; i<=k; i++){
                // 1호는 어차피 1명이므로 2호부터 시작
                for (int j = 2; j<=n; j++){
                    floor[0][j] = floor[0][j-1] + floor[0][j];
                }
            }
            System.out.println(floor[0][n]);
        }
        scanner.close();
    }
}



